Collections/collections_home_task.py

```diff
@@ -0,0 +1,71 @@
+"""
+Home task for Module 2 - Collections
+
+Write a code, which will:
```

Comment:
```suggestion
import random
import string

# create list of random dicts
random_dicts = [{random.choice(string.ascii_lowercase): random.randint(0, 100) for _ in range(random.randint(1, 100))}
                for _ in range(random.randint(2, 10))]
print(f'List of random dictionaries: {random_dicts}.')

"""
An idea here is to create structure which will store all options of (index, value) for each letter in one place.
It will be easy to process such thing: {'a': [(0, 15)], 'b': [(0, 3), (1, 17)]}. It tells that there is one 'a' letter 
in 0 dict with value 15 and two 'b' letters: in 0 dict with value 3; in 1 dict with value 17
"""
letters_to_combinations_list = {}  # combination is a tuple of (index, value)
# enumerate allows create indexes in-place (instead of manually creating index). useful function, please read about it
for index, dictionary in enumerate(random_dicts):
    for letter, value in dictionary.items():
        if letter not in letters_to_combinations_list:
            letters_to_combinations_list[letter] = [(index, value)]  # create new key and assign initial value
        else:
            letters_to_combinations_list[letter].append((index, value))  # append new combination to list
print(letters_to_combinations_list)

# Now when we have all combinations in one place we should just compare them and create dict accordingly to requirements
result = {}
for letter, combinations in sorted(letters_to_combinations_list.items()):  # sorted for showing in alphabet order
    index, max_value = combinations[0]  # we take first combination and assume that it's the max value
    for combination in combinations[1:]:  # now we go from 1 tuple (0 we already assigned)
        if combination[1] > max_value:  # if value of current combination more than we assigned let's reassign them
            index, max_value = combination
    result[f'{letter}_{index}'] = max_value  # now when we know max value we can create a new pair in result dict
print(result)

```

Reply:
I added there my version of the code how it could look like, just for your acquaintance. You can take couple of ideas and examples for your future work. The same for comments, they should help reader to undertstand the code, guide him through it.